liblangutil/Exceptions.cpp

```diff
@@ -62,7 +77,20 @@ Error::Error(
 	case Type::TypeError:
 		m_typeName = "TypeError";
 		break;
-	case Type::Warning:
+	case Type::CompilerError:
+		m_typeName = "CompilerError";
+		break;
+	case Type::UnimplementedFeatureError:
+		m_typeName = "UnimplementedFeatureError";
+		break;
+	case Type::YulException:
+		m_typeName = "YulException";
+		break;
+	case Type::SMTLogicException:
+		m_typeName = "SMTLogicException";
+		break;
+	default:
+		solAssert(_type == Type::Warning);
```

Comment:
Does it possibly make sense to just use a map here instead of a long switch? 

Either way, can't we use `formatErrorType` and reuse that logic instead of recreating it in two places?

Reply:
`switch` is nice in cases when there's no `default`. The compiler can tell you that you need a new branch when stuff is added to the enum.

Reply:
The `switch case` statement here seems to be identical to the one in method `formatErrorType`. Couldn't we use it here instead?

Reply:
Been thinking of doing that myself. Will implement that and push changes.

Reply:
Been considering doing that myself. Will implement it in the next push